bundles/org.eclipse.swt/Build-SWT-native-binaries-for-running-platform.launch

```diff
@@ -16,9 +16,10 @@
     <stringAttribute key="M2_USER_SETTINGS" value=""/>
     <booleanAttribute key="M2_WORKSPACE_RESOLUTION" value="false"/>
     <booleanAttribute key="org.eclipse.debug.core.ATTR_FORCE_SYSTEM_CONSOLE_ENCODING" value="false"/>
+    <stringAttribute key="org.eclipse.debug.core.ATTR_REFRESH_SCOPE" value="${workspace}"/>
```

Comment:
That's a good idea, but instead of refreshing the entire workspace, which can take quite some time if many more than only the SWT projects are contained just refreshing the native SWT fragments would be more precise.
Of course it would have the drawback that if new fragments arrive there is a risk that they go out of sync.

```suggestion
    <stringAttribute key="org.eclipse.debug.core.ATTR_REFRESH_SCOPE" value="${working_set:&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#13;&#10;&lt;resources&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.cocoa.macosx.aarch64&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.cocoa.macosx.x86_64&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.gtk.linux.aarch64&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.gtk.linux.loongarch64&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.gtk.linux.ppc64le&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.gtk.linux.x86_64&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.win32.win32.aarch64&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.win32.win32.x86_64&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;/resources&gt;}"/> 
```

Ideally only the fragment for which the natives have been built would be refreshed (this would also go out of sync as long as this launch config works in general).
```suggestion
    <stringAttribute key="org.eclipse.debug.core.ATTR_REFRESH_SCOPE" value="${working_set:&lt;?xml version=&quot;1.0&quot; encoding=&quot;UTF-8&quot;?&gt;&#13;&#10;&lt;resources&gt;&#13;&#10;&lt;item path=&quot;/org.eclipse.swt.${target.ws}.${target.os}.${target.arch}&quot; type=&quot;4&quot;/&gt;&#13;&#10;&lt;/resources&gt;}"/>
```
But I don't know if variables are resolved there.
Could you check that? I don't have a quick way to do that because on Windows the native hooks refresh automatically and I see no error when running this build.

Reply:
The 1st approach works:

![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/2205684/e90459de-5a9c-40e2-8cb9-2d7bfe170120)

The 2nd one doesn't:

![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/2205684/bfe27993-3167-465a-af9e-9cfdad9d87da)

Committing the 1st one